samza-core/src/main/java/org/apache/samza/processor/StreamProcessor.java

```diff
@@ -401,11 +403,29 @@ public void onJobModelExpired() {
       public void onNewJobModel(String processorId, JobModel jobModel) {
         synchronized (lock) {
           if (state == State.IN_REBALANCE) {
+            containerStartLatch = new CountDownLatch(1);
             containerShutdownLatch = new CountDownLatch(1);
             container = createSamzaContainer(processorId, jobModel);
             container.setContainerListener(new ContainerListener());
             LOGGER.info("Starting the container: {} for the stream processor: {}.", container, processorId);
             containerExcecutorService.submit(container);
+
+            try {
```

Comment:
Consider this: Move try inside if and then move:

`state = State.STOPPED;
jobCoordinator.stop();
processorListener.afterFailure(new TimeoutException("Container start timed out after "+taskStartMs+ " ms"));`

inside a finally block